mathics/core/evaluators.py

```diff
@@ -37,20 +43,32 @@ def apply_N(expression, evaluation, prec=SymbolMachinePrecision):
     return result.evaluate(evaluation)
 
 
-def apply_nvalues(expr, prec, evaluation):
+def apply_nvalues(
```

Comment:
I appreciate and like the comments added below. I feel that somehow there is some higher-level idea that we might be missing.  So I'd just suggest a comment above this function to look over again at some future point. 

For example: 

```
# TODO:  Revisit - can this be simplified? Is some broader framework this fits into?

Reply:
Done.